### ISSUE TITLE
Incorrect "undefined variable" error for class variables as return type annotations

### ISSUE THREAD

**Issue by nicktimko** [PyCQA/pylint#1976]
pylint seems to incorrectly identify an undefined variable error when using a class variable as a *return* type annotation. Argument type annotations appear to be OK.

### Steps to reproduce
```py
# lintcheck.py
class MyObject:
    class MyType:
        pass
    def my_method(self) -> MyType:
        pass
```

### Current behavior
```
$ pylint lintcheck.py -E
No config file found, using default configuration
************* Module lintcheck
E:  5,27: Undefined variable 'MyType' (undefined-variable)
```

### Expected behavior

No error.

Curiously, the below doesn't emit any errors, where `MyType` should have identical scope:

```python
# lintcheck_ok.py
class MyObject2:
    class MyType:
        pass
    def my_method(self, x: MyType):
        pass
    def my_thing(self, x=MyType):
        pass
```

### pylint --version output

```sh
$ pylint --version
No config file found, using default configuration
pylint 1.8.3, 
astroid 1.6.2
Python 3.6.4 (default, Mar 26 2018, 16:21:07) 
[GCC 5.4.0 20160609]
```

**Comment by PCManticore** [comment]
Thanks, I can reproduce the issue!

**Comment by pt12lol** [comment]
Another test case: https://stackoverflow.com/questions/61271870/methods-decorator-parameter-cannot-be-a-list-comprehension-based-on-class-stat

**Comment by zhiltsov-max** [comment]
The same thing for 
```
class Project:
    MyType = NewType('MyType', str)

    def foo(self, obj: MyType):                    # Undefined variable 'MyType'
```
